app/routes/goal_routes.py

```diff
@@ -0,0 +1,114 @@
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+from flask import abort, Blueprint, jsonify, make_response, request
+from .helpers_routes import validate_obj
+
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+
+@goals_bp.route("", methods=["GET"])
+def read_all_goals():
+    goals = Goal.query.all()
+
+    goals_response = [goal.to_dict() for goal in goals]
+
+    return jsonify(goals_response)
+
+
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def get_one_goal(goal_id):
+    chosen_goal = validate_obj(Goal, goal_id)
+
+    response = {
+        "goal": chosen_goal.to_dict()
+    }
+
+    return jsonify(response)
+
+
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+
+    if request_body == {}:
+        return jsonify({"details": "Invalid data"}), 400
```

Comment:
you should be checking to see if the request body lacks a title.  You should also indicate what/why the data is invalid.